lib/block-supports/layout.php

```diff
@@ -150,12 +157,22 @@ function gutenberg_get_layout_style( $selector, $layout, $has_block_gap_support
 			);
 		}
 
-		if ( $has_block_gap_support ) {
-			if ( is_array( $gap_value ) ) {
-				$gap_row    = isset( $gap_value['top'] ) ? $gap_value['top'] : $fallback_gap_value;
-				$gap_column = isset( $gap_value['left'] ) ? $gap_value['left'] : $fallback_gap_value;
-				$gap_value  = $gap_row === $gap_column ? $gap_row : $gap_row . ' ' . $gap_column;
+		if ( $has_block_gap_support && isset( $gap_value ) ) {
+			$combined_gap_value = '';
+			$gap_sides          = is_array( $gap_value ) ? array( 'top', 'left' ) : array( 'top' );
+
+			foreach ( $gap_sides as $gap_side ) {
+				$process_value = is_string( $gap_value ) ? $gap_value : _wp_array_get( $gap_value, array( $gap_side ), $fallback_gap_value );
```

Comment:
It looks like `_wp_array_get` is returning `null` for `null` values.

So, an incoming `$gap_value` of `array(2) { ["top"]=> NULL ["left"]=> string(4) "38px" } `

Will result in a `$combined_gap_value` of `38px`.

Steps to reproduce:

1. Insert social links block
2. Add a block gap value for one side only. 

The editor produces the following CSS:

```css
    gap: var(--wp--preset--spacing--20) 92px;
```

The frontend

```css
gap: 38px;
```

Reply:
Sorry, ignore this. I refreshed the editor and I can't reproduce any more.

Reply:
🤭

Reply:
Okay, cool. We can always do more detailed testing with this behaviour once we've got the spacing controls in there, I imagine it'll make testing a little easier since we'll be doing less manual fiddling with markup values 🤔

Thanks for all the testing here!